redisvl/index.py

```diff
@@ -169,78 +139,62 @@ class SearchIndexBase:
 
     def __init__(
         self,
-        name: str,
-        prefix: str = "rvl",
-        storage_type: str = "hash",
-        key_separator: str = ":",
-        fields: Optional[List["Field"]] = None,
+        schema: Schema,
+        redis_url: Optional[str] = None,
+        connection_args: Dict[str, Any] = {},
         **kwargs,
     ):
-        """Initialize the RedisVL search index class.
-
-        Args:
-            name (str): Index name.
-            prefix (str, optional): Key prefix associated with the index.
-                Defaults to "rvl".
-            storage_type (str, optional): Underlying Redis storage type (hash
-                or json). Defaults to "hash".
-            key_separator (str, optional): : Separator character to combine
-                prefix and key value for constructing redis keys.
-                Defaults to ":".
-            fields (Optional[List[Field]], optional): List of Redis fields to
-                index. Defaults to None.
+        """Initialize the RedisVL search index class with a schema,
+           redis_url, connection_args, and other kwargs.
         """
-        self._name = name
-        self._prefix = prefix
-        self._key_separator = key_separator
-        self._storage_type = StorageType(storage_type)
-        self._fields = fields
-
-        # configure storage layer
-        self._storage = self.STORAGE_MAP[self._storage_type](  # type: ignore
-            self._prefix, self._key_separator
+        # final validation on schema object
+        if not schema or not isinstance(schema, Schema):
```

Comment:
pass ``**kwargs`` to the schema init so it's not breaking from 0.0.5?

Reply:
ensuring that we can pass it the existing init method paramteres and it still works. breaking the init method even in a minor release is tough for users.